lld/test/COFF/timestamp.test

```diff
@@ -3,9 +3,11 @@ RUN: yaml2obj %p/Inputs/generic.yaml -o %t.obj
 RUN: lld-link %t.obj /debug /Brepro /entry:main /nodefaultlib /out:%t.1.exe
 RUN: lld-link %t.obj /debug /Brepro /entry:main /nodefaultlib /out:%t.2.exe
 RUN: lld-link %t.obj /debug /timestamp:0 /entry:main /nodefaultlib /out:%t.3.exe
+RUN: env SOURCE_DATE_EPOCH=0 lld-link %t.obj /debug /entry:main /nodefaultlib /out:%t.4.exe
```

Comment:
It'd be useful to test another value like `clang/test/Preprocessor/SOURCE_DATE_EPOCH.c`

Reply:
Thanks, I added more tests for more interesting cases.

What do you think of backporting this to 18.x? It’s not a bug fix per se, but a pretty safe improvement.

Reply:
LGTM for a backport to improve reproducibility! 

https://reproducible-builds.org/